sycl/include/syclcompat/device.hpp

```diff
@@ -309,6 +309,14 @@ class device_ext : public sycl::device {
     _saved_queue = _default_queue = _queues.front().get();
   }
 
+  void set_default_queue(const sycl::queue &q) {
+    std::lock_guard<std::mutex> lock(m_mutex);
+    _queues[0] = std::make_shared<sycl::queue>(q);
```

Comment:
I think you may want to flush the previous `default_queue` here to make sure that no work is left on it

Reply:
You are right. I've added a synchronization point for the previous queue.